module/Finna/src/Finna/RecordDriver/SolrLido.php

```diff
@@ -384,6 +384,38 @@ public function getAlternativeTitles()
         return $results;
     }
 
+    /**
+     * Get an array of related publications for the record.
+     *
+     * @return array
+     */
+    public function getRelatedPublications()
+    {
+        $results = [];
+        foreach ($this->getXmlRecord()->xpath(
+            'lido/descriptiveMetadata/objectRelationWrap/relatedWorksWrap/'
+            . 'relatedWorkSet'
+        ) as $node) {
+            if (!empty($node->relatedWork->displayObject)) {
+                $title = (string)$node->relatedWork->displayObject;
+                $attributes = $node->relatedWork->displayObject->attributes();
+                $label = !empty($attributes->label)
+                    ? (string)$attributes->label : '';
+                $term = !empty($node->relatedWorkRelType->term)
+                    ? (string)$node->relatedWorkRelType->term : '';
+                if (($label)
```

Comment:
Huomasin vasta nyt, että tuossa on vähän ehdoissa toistoa ja ylimääräisiä sulkuja. compact ei myöskään ole mitenkään pakollinen, kun avain ja muuttuja eivät vastaa toisiaan. Voinee siis laittaa näin:
```
                if (in_array($term, ['kirjallisuus', 'lähteet'])) {
                    $results[] = [
                        'title' => $title,
                        'label' => $label ?: $term
                    ];
                }

```